src/Protocol.h

```diff
@@ -122,16 +122,20 @@ struct HeaderPrefix {
 
 /**
  * Describes the wire format for header fields that are common to all packet
- * types.
+ * types. Note: the first 4 bytes are identical for TCP, UDP, and Homa.
  */
 struct CommonHeader {
+    uint16_t sport, dport;///< Transport layer (L4) source and destination ports
+                          ///< in network byte order; only used by DataHeader.
```

Comment:
If they are only used by the `DataHeader` why are they in the `CommonHeader`?

Reply:
Because they need to be placed in the first 4 bytes. Besides, some of the other headers like `GrantHeader` could also benefit from `sport/dport` if we want to take advantage of RSS to route GRANTs directly to the CPU processing the data packets of the same message.